src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -744,12 +861,79 @@ private boolean handleMaterials(@NotNull final Block block, @NotNull final IBloc
             return true;
         }
 
-        if (isBlockFree(block, block.getMetaFromState(blockState)))
+        final List<ItemStack> itemList = new ArrayList<>();
+        itemList.add(BlockUtils.getItemStackFromBlockState(blockState));
+        if(job instanceof JobBuilder && ((JobBuilder) job).getStructure() != null
+                && ((JobBuilder) job).getStructure().getBlockInfo() != null && ((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
         {
-            return true;
+            itemList.addAll(getItemStacksOfTileEntity(((JobBuilder) job).getStructure().getBlockInfo().tileentityData));
```

Comment:
we should not hardcode the builder in an ai, that is wrong on many levels, for example for the miner it breaks stuff